### PR TITLE
Canvas: Button API Editor support template variables

### LINKED ISSUE #74224: Canvas: Button API Editor support template variables
Support using template variables in button api editor. Most likely these will be useful in the url / parameters / payload

### PR DESCRIPTION
**Pull request by adela-almasan**
Adds support for template variables in URL, parameters and body.

https://github.com/grafana/grafana/assets/88068998/0c03dc7e-dbf0-4195-bf52-026f4ba318ae

Fixes #74224 

**Special notes for your reviewer:**
#74220 needs to be merged first.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/canvas/elements/icon.tsx
Comment (adela-almasan) [inline public/app/features/canvas/elements/icon.tsx:11]: Is this really needed/supported for icon?

Comment (ryantxu) [reply public/app/features/canvas/elements/icon.tsx:11]: looks like it is for the onClick handler 🤷🏻

Comment (ryantxu) [reply public/app/features/canvas/elements/icon.tsx:11]: at this stage, i would keep it out of icon -- i *think* it existed as a cheap button.

So i vote we remove it for now

Comment (adela-almasan) [reply public/app/features/canvas/elements/icon.tsx:11]: _cheap button_ :sweat_smile: I'll remove it for now, we'll bring it back if needed.

File: public/app/features/canvas/elements/icon.tsx
Comment (nmarrs) [inline public/app/features/canvas/elements/icon.tsx:49]: Is there any reason to not support api editor functionality for the icon element as well? This has been a feature request that has come up [a few times](https://github.com/grafana/grafana/discussions/56835?sort=new#discussioncomment-3918128). We should probably disable the onclick call though when inline editing is enabled (similar to button)

Comment (nmarrs) [reply public/app/features/canvas/elements/icon.tsx:49]: at very least we can create a follow-up issue to consider re-introducing support for this once we get button element in a good place

File: public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx
Comment (nmarrs) [inline public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx:1]: It seems like this PR may be blocked by figuring out / merging in the [support adding parameters PR](https://github.com/grafana/grafana/issues/74220)

File: public/app/plugins/panel/canvas/editor/element/APIEditor.tsx
Comment (nmarrs) [inline public/app/plugins/panel/canvas/editor/element/APIEditor.tsx:216]: It might be nice if we can figure out how to add the autocomplete functionality (i.e. when typing ${ the list of possible template variables come up) - I think I've seen this functionality somewhere else in our UI before but not sure exactly where 🤷

**Review by nmarrs — COMMENTED** [review]
This is working well! I think this may be blocked until we get the support [adding params PR merged](https://github.com/grafana/grafana/issues/74220)

Also for icon api support we can remove that for now, but should create a follow up issue to investigate adding it back once button is in a good place

File: public/app/features/canvas/elements/icon.tsx
Comment (adela-almasan) [reply public/app/features/canvas/elements/icon.tsx:49]: The editor was commented out so the functionality was not exposed. Ryan also had some comments about it. I can create a follow up issue to bring it back if we decide there's value in that..

File: public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx
Comment (nmarrs) [inline public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx:1]: I think we can remove this file (which will also cleanup associated betterer issues)

File: public/app/plugins/panel/canvas/editor/element/APIEditor.tsx
Comment (nmarrs) [reply public/app/plugins/panel/canvas/editor/element/APIEditor.tsx:57]: Tested this a bit locally and it seems that the interpolation breaks when canvas has inline editing mode disabled in both dashboard view and panel edit view 🤔 

https://github.com/grafana/grafana/assets/22381771/6a611360-729f-4099-9474-21de43c11017

File: public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx
Comment (adela-almasan) [reply public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx:1]: oops, merging conflicts left over!

File: public/app/plugins/panel/canvas/editor/element/APIEditor.tsx
Comment (adela-almasan) [reply public/app/plugins/panel/canvas/editor/element/APIEditor.tsx:57]: hm, how is that related? 🤔 I'll take a look..

**Review by nmarrs — APPROVED** [review]
🚀

**Pull request merged**
